ssh/tailssh/incubator.go

```diff
@@ -233,11 +217,75 @@ func beIncubator(args []string) error {
 
 	euid := os.Geteuid()
 	runningAsRoot := euid == 0
-	if runningAsRoot && ia.loginCmdPath != "" {
-		// Check if we can exec into the login command instead of trying to
-		// incubate ourselves.
-		if la := ia.loginArgs(); la != nil {
-			return unix.Exec(ia.loginCmdPath, la, os.Environ())
+
+	// Decide whether we should attempt to get a full login shell using either
```

Comment:
Previously, the logic of when to attempt `login` and which command to use was spread across two processes and three functions. This consolidates it all to one place to make it easier to follow.